stl/inc/xmemory

```diff
@@ -23,6 +23,33 @@ _STL_DISABLE_CLANG_WARNINGS
 #undef new
 
 _STD_BEGIN
+// STRUCT TEMPLATE _Tidy_guard
+template <class _Ty>
+struct _Tidy_guard { // class with destructor that calls _Tidy

```

Comment:
Pre-existing: should these guards be noncopyable? (Or did that interfere with aggregate initialization?) Also, should we mark them as `TRANSITION, P1771R1 [[nodiscard]] For Constructors` for when we work on #63?

Reply:
Yes, that interferes with aggregate initialization as of C++20. (They used to be noncopyable but we had to back that out when p1008r1 happened)

Reply:
I don't think giving up aggregate init is worth a [[nodiscard]] here.

Reply:
Ah, ok.

Reply:
> I don't think giving up aggregate init is worth a [[nodiscard]] here.

Don't forget that `[[nodiscard]]` can be applied to types as well as functions; maybe `struct _NODISCARD _Tidy_guard` is the right answer?

Reply:
That would complain if someone returned such a type from a function and then dropped it on the floor, but it wouldn’t complain about `_Tidy_guard{this}` which is the danger I’m concerned about.

Reply:
MSVC warns in that case: https://godbolt.org/z/Jr7PQa.